scripts/start_port.sh

```diff
@@ -67,8 +67,24 @@ echo "Applying automated fixes"
     sed -i 's/Mathbin\./Mathlib\./g' "$mathlib4_path"
     sed -i '/^import/{s/[.]Gcd/.GCD/g; s/[.]Modeq/.ModEq/g; s/[.]Nary/.NAry/g; s/[.]Peq/.PEq/g; s/[.]Pfun/.PFun/g; s/[.]Pnat/.PNat/g; s/[.]Smul/.SMul/g; s/[.]Zmod/.ZMod/g}' "$mathlib4_path"
 
-    # awk script taken from https://github.com/leanprover-community/mathlib4/pull/1523
-    awk '{do {{if (match($0, "^  by$") && length(p) < 98 && (!(match(p, "^[ \t]*--.*$")))) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1) if (getline==0) print p}' "$mathlib4_path" > "$mathlib4_path.tmp"
+    python3 - << BYBACK
+from collections import deque
+lns = deque([], 2)
+with (open("$mathlib4_path", "r", encoding="utf-8", newline="\n") as f,
+      open("$mathlib4_path.tmp", "w", encoding="utf-8", newline="\n") as g):
+    for ln_raw in f:
+        ln = ln_raw.strip("\n")
+        lns.append(ln)
+        if len(lns) > 1 and lns[1] == "  by" and len(lns[0]) < 98 and not lns[0].lstrip().startswith("--"):
+            lns.pop()
+            lns[0] += " by"
+        elif len(lns) > 1:
+            print(lns[0], file=g)
+    lns.popleft()
+    for ln in lns:
+        print(ln, file=g)
+BYBACK
```

Comment:
Can you move this to `fix-line-breaks.py` so that it's next to `fix-line-breaks.sh`?

Reply:
Done, though I am not entirely sure whether the method of getting the new Python file's path is fully cross-platform.